planner/core/logical_plan_builder.go

```diff
@@ -2041,17 +2041,80 @@ func getUintFromNode(ctx sessionctx.Context, n ast.Node) (uVal uint64, isNull bo
 	return 0, false, false
 }
 
+// getUintFromLimitNode gets uint64 value from ast.Node.
+// For ordinary statement, node should be uint64 constant value.
+// For prepared statement, node is string. We should convert it to uint64.
+func getUintFromLimitNode(ctx sessionctx.Context, n ast.Node) (uVal uint64, isNull bool, isExpectedType bool) {
+	var val interface{}
+	switch v := n.(type) {
+	case *driver.ValueExpr:
+		val = v.GetValue()
+	case *driver.ParamMarkerExpr:
+		if !v.InExecute {
+			return 0, false, true
+		}
+		if expected := checkParamTypeExpected(v); !expected {
+			return 0, false, false
+		}
+		param, err := expression.ParamMarkerExpression(ctx, v, false)
+		if err != nil {
+			return 0, false, false
+		}
+		str, isNull, err := expression.GetStringFromConstant(ctx, param)
+		if err != nil {
+			return 0, false, false
+		}
+		if isNull {
+			return 0, true, true
+		}
+		val = str
+	default:
+		return 0, false, false
+	}
+	switch v := val.(type) {
+	case uint64:
+		return v, false, true
+	case int64:
+		if v >= 0 {
+			return uint64(v), false, true
+		}
+	case string:
+		sc := ctx.GetSessionVars().StmtCtx
+		uVal, err := types.StrToUint(sc, v, false)
+		if err != nil {
+			return 0, false, false
+		}
+		return uVal, false, true
+	}
```

Comment:
Should we add `case int` here?

Reply:
I think the `Datum` 's value are set and get by unify functions, but there seems no `SetInt()` or `GetInt()` function exist, and the `Datum` struct has no `int` type members except `interface{}` can be used to save。So i guess there is no need to `case int `

Reply:
And this function is clone from `getUintFromNode`, this function has also no `int` case